diff --git a/dest/contract_function_simulator/oracle/legacy_oracle_mappings.js b/dest/contract_function_simulator/oracle/legacy_oracle_mappings.js
--- a/dest/contract_function_simulator/oracle/legacy_oracle_mappings.js
+++ b/dest/contract_function_simulator/oracle/legacy_oracle_mappings.js
@@ -31,6 +31,9 @@
             ]),
         utilityGetL1ToL2MembershipWitness: (contractAddress, messageHash, secret)=>oracle.aztec_utl_getL1ToL2MembershipWitness(contractAddress, messageHash, secret),
         utilityEmitOffchainEffect: (data)=>oracle.aztec_utl_emitOffchainEffect(data),
+        utilityFetchTaggedLogs: (pendingTaggedLogArrayBaseSlot)=>oracle.aztec_utl_getPendingTaggedLogs(pendingTaggedLogArrayBaseSlot, [toACVMField(AztecAddress.ZERO)]),
+        utilityValidateAndStoreEnqueuedNotesAndEvents: (contractAddress, noteValidationRequestsArrayBaseSlot, eventValidationRequestsArrayBaseSlot, maxNotePackedLen, maxEventSerializedLen)=>oracle.aztec_utl_validateAndStoreEnqueuedNotesAndEvents(contractAddress, noteValidationRequestsArrayBaseSlot, eventValidationRequestsArrayBaseSlot, maxNotePackedLen, maxEventSerializedLen, [toACVMField(AztecAddress.ZERO)]),
+        utilityBulkRetrieveLogs: (contractAddress, logRetrievalRequestsArrayBaseSlot, logRetrievalResponsesArrayBaseSlot)=>oracle.aztec_utl_getLogsByTag(contractAddress, logRetrievalRequestsArrayBaseSlot, logRetrievalResponsesArrayBaseSlot, [toACVMField(AztecAddress.ZERO)]),
         // Renames (same signature, different oracle name)
         privateNotifySetMinRevertibleSideEffectCounter: (counter)=>oracle.aztec_prv_notifyRevertiblePhaseStart(counter),
         privateIsSideEffectCounterRevertible: (sideEffectCounter)=>oracle.aztec_prv_isExecutionInRevertiblePhase(sideEffectCounter),
